dnsmasq/Dockerfile

```diff
@@ -16,4 +16,5 @@ FROM alpine:3.4
 MAINTAINER Girish Kalele <gkalele@google.com>
 RUN apk --no-cache add dnsmasq
 COPY dnsmasq.conf /etc/dnsmasq.conf
+STOPSIGNAL SIGCONT
```

Comment:
Hai guys, 
don't we have a feature for exactly this called pre-stop hook? I mean: 
```yaml
lifecycle:
  preStop:
    exec:
      command:
      - /bin/sh
      - -ec
      - |
        echo "sleeping 60s"
        sleep 60
```

The pre-stop duration is taken from deletionGrace, meaning if you set `terminationGrace` to 60s and `sleep 60s` in your pre-stop, you will receive a sigkill in 60s. Prestop should be called *before* you get the `docker stop`, ie before the SIGTERM. 

This should work cross container runtime, maybe I misunderstood what you're trying to do?